### ISSUE TITLE
Create a CLI

### ISSUE THREAD

**Issue by jovandeginste** [drakkan/sftpgo#23]
The REST API is great, of course, but a more intuitive CLI would also be awesome.

Two options:

* build a separate tool (in any language, but I personally would prefer golang)
* make the daemon also the CLI (see eg. docker, vault and nomad)

In the first scenario, you will need to keep both projects in sync where relevant.

In the second case, you should use a specific "command" argument to run the server, so you don't accidentally run it and create all kinds of default files (log, id_rsa)

eg.:
```
$ sftpgo serve --config-dir=/etc/sftpgo/
$ sftpgo user create foo --uid=1000 --homedir=/var/lib/sftpgo/foo --keys=@/path/to/authorized_keys --no-password
```

**Comment by drakkan** [comment]
I just added a CLI tool in python

https://github.com/drakkan/sftpgo/tree/master/scripts

If future we can try to make the daemon a CLI tool in this case we need a more advanced arguments parser library

**Comment by drakkan** [comment]
Of course if you can send a pull request it would be accepted, simply this is not a top priority for me now.

If you want to make a such CLI the sub commands should be named as the matching REST API operationId and the parameters should match the ones defined in REST API, thanks!

**Comment by jovandeginste** [comment]
I will wait with any work here until you decide which CLI library you will use.

**Comment by jovandeginste** [comment]
I already know why the CLI should not be in Python :-) Need it on a CentOS 7 server:

```text
$ ./sftpgo_api_cli.py --help
/usr/bin/env: python3: No such file or directory
[edited python3 -> python]
$ ./sftpgo_api_cli.py 
Traceback (most recent call last):
  File "./sftpgo_api_cli.py", line 5, in <module>
    import requests
ImportError: No module named requests
$ sudo yum -y install python-requests
[...]
$ ./sftpgo_api_cli.py --help
Traceback (most recent call last):
  File "./sftpgo_api_cli.py", line 7, in <module>
    import urllib.parse as urlparse
ImportError: No module named parse
```

I gave up at that moment :-)

**Comment by drakkan** [comment]
python2 should be now supported

https://github.com/drakkan/sftpgo/commit/4009a9371329b4f012e56e1913fb0e78a39f239b

anyway I tested python 2.7.16 not the version shipped with centos 7, please let me know it it works there too

**Comment by jovandeginste** [comment]
Works indeed!

Just FYI, but you can create some clients with this script:

```bash
# Generates bash script in $(PWD)/bash/
$ docker run -v $(pwd):/out --rm openapitools/openapi-generator-cli generate -i https://raw.githubusercontent.com/drakkan/sftpgo/master/api/schema/openapi.yaml -g bash -o /out/bash/
```

**Comment by drakkan** [comment]
I used `swagger-codegen` before, I didn't know this generator, thanks for sharing :)

I'll add some info in the README later

**Comment by drakkan** [comment]
I'm quite happy with the current sftpgo python CLI:

- on Linux it has basically zero dependency since python is installed by default and `python-requests` is available in the main repo for all major distro
- it is quite easy to maintain and extend. Think for example for digest auth: in python is supported by `requests`, in Go we have to select and test an external dependency. It is doable in Go too but it requires a bit more effort
- If someone want to use bash scripts they can be autogenerated using one of the available openapi generator, and, if needed/asked, I can add some examples using `curl` to the docs too

So basically the CLI already exists and I don't see the need to rewrite it in Go.

Anyway if someone would like to send a pull request for an integrated CLI, with at least the same features as the current python CLI, I will be glad to merge it. We are using cobra now and add additional subcommands to the `sftpgo` executable should be quite easy :)
